src/Schema/SqliteSchemaManager.php

```diff
@@ -176,7 +176,7 @@ public function listTableForeignKeys($table, $database = null)
                 $id = $value['id'];
 
                 $tableForeignKeys[$key] = array_merge($tableForeignKeys[$key], [
-                    'constraint_name' => isset($names[$id]) && $names[$id] !== '' ? $names[$id] : $id,
+                    'constraint_name' => isset($names[$id]) && $names[$id] !== '' ? $names[$id] : null,
```

Comment:
fixes https://dbfiddle.uk/?rdbms=sqlite_3.27&fiddle=4651f0cc575a2f2df11856003d2e0af8

using `$id` as a fallback name was wrong, as name cannot start with a number

we must use `null` as a fallback name, so `CONSTRAINT <name> FOREIGN KEY...` is not used for `CREATE TABLE`

https://www.sqlite.org/syntax/table-constraint.html

Reply:
This actually needs https://github.com/doctrine/dbal/pull/5490, so fixes for it are not part of this PR.